sonarts-core/src/rules/noRedundantParenthesesRule.ts

```diff
@@ -34,20 +35,19 @@ export class Rule extends Lint.Rules.AbstractRule {
   };
 
   public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
-    return this.applyWithWalker(new Walker(sourceFile, this.getOptions()));
+    return new Visitor(this.getOptions().ruleName).visit(sourceFile).getIssues();
   }
 }
 
-class Walker extends Lint.RuleWalker {
+class Visitor extends SonarRuleVisitor {
   private static readonly MESSAGE = "Remove these useless parentheses.";
 
-  visitNode(node: ts.Node): void {
-    if (is(node, ts.SyntaxKind.ParenthesizedExpression)) {
-      const parenthesizedExpression = node as ts.ParenthesizedExpression;
-      if (is(parenthesizedExpression.expression, ts.SyntaxKind.ParenthesizedExpression)) {
-        this.addFailureAtNode(node, Walker.MESSAGE);
-      }
+  public visitParenthesizedExpression(node: ts.ParenthesizedExpression) {
+    const parenthesizedExpression = node as ts.ParenthesizedExpression;
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove this unnecessary cast. ([typescript:S4325](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=typescript%3AS4325))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=sonarts&branch=sonar-issue&issues=AWCd631y_Q-0dz4JAbfK&open=AWCd631y_Q-0dz4JAbfK)

[ISSUE_KEY]: AWCd631y_Q-0dz4JAbfK